service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java

```diff
@@ -70,7 +72,14 @@ protected void processException(final RoutingContext ctx, final Throwable except
 
     @Override
     public void parseCredentials(final RoutingContext context, final Handler<AsyncResult<JsonObject>> handler) {
-        super.parseCredentials(context, ar -> {
+        final RoutingContextDecorator routingContextDecorator = new RoutingContextDecorator(context.currentRoute(),
+                context) {
+
+            public void fail(final Throwable throwable) {
+                context.fail(HttpURLConnection.HTTP_UNAUTHORIZED, throwable);
```

Comment:
wouldn't be sufficient to add this line in an `else` block in the handler below? Using the `RoutingContextDecorator` should probably work but I think we can achieve the same result with a little less *ammo*, don't you think?
Can you also add a corresponding unit test?

Reply:
@sophokles73 Already I checked for the `else` block, it doesn't work. Vertx `BasicAuthHandlerImpl` invokes `context.fail(e)` when error occurs, thereby the `DefaultFailureHandler` is being invoked.

Reply:
I see. Can you then please add a comment explaining why using the `RoutingContextDecorator` in this case is necessary?

Reply:
@sophokles73 I have added a unit test and a comment for the above.